src/bot/SoundBot.ts

```diff
@@ -43,53 +43,55 @@ export default class SoundBot extends Client {
   private addEventListeners() {
     this.on('ready', this.onReady);
     this.on('message', this.onMessage);
-    this.on('voiceStateUpdate', this.onUserLeavesVoiceChannel);
-    this.on('voiceStateUpdate', this.onUserJoinsVoiceChannel);
+    this.on('voiceStateUpdate', this.handleVoiceStateUpdate);
     this.on('guildCreate', this.onBotJoinsServer);
   }
 
   private onReady() {
-    if (!this.user) {
-      throw new Error('User not set after ready.');
-    }
-    this.user.setActivity(this.config.game);
-    this.commands.registerUserCommands(this.user);
+    this.user!.setActivity(this.config.game);
+    this.commands.registerUserCommands(this.user!);
   }
 
-  private onUserJoinsVoiceChannel(prevState: GuildMember, user: GuildMember) {
-    if (!this.user || user.id === this.user.id) return;
+  private handleVoiceStateUpdate(oldState: VoiceState, newState: VoiceState) {
+    if (!oldState.channelID && newState.channelID) {
+      return this.onUserJoinsVoiceChannel(newState.channel!, newState.member!.user);
+    }
 
-    if (!user.voice) return;
-    if (!entrances.exists(user.id)) return;
+    if (oldState.channelID && !newState.channelID) {
+      return this.onUserLeavesVoiceChannel(oldState.channel!, oldState.member!.user);
+    }
+  }
+
+  private onUserJoinsVoiceChannel(channel: VoiceChannel, user: User) {
+    if (!entrances.exists(user.id)) {
+      return;
+    }
 
     const sound = entrances.get(user.id);
-    if (!getSounds().includes(sound)) return;
+    if (!getSounds().includes(sound)) {
+      return;
+    }
 
-    const { channel } = user.voice;
-    if (!channel) return;
     this.queue.add(new QueueItem(sound, channel));
   }
 
-  private onUserLeavesVoiceChannel(prevState: GuildMember, user: GuildMember) {
-    if (!this.user || user.id === this.user.id) return;
-
-    if (!prevState.voice || prevState.voice.channelID === user.voice.channelID) return;
-    if (!exits.exists(user.id)) return;
+  private onUserLeavesVoiceChannel(channel: VoiceChannel, user: User) {
+    if (!exits.exists(user.id)) {
+      return;
+    }
 
     const sound = exits.get(user.id);
     if (!getSounds().includes(sound)) return;
 
-    const { channel } = prevState.voice;
-    if (!channel) return;
-    this.queue.add(new QueueItem(sound, channel));
+    this.queue.add(new QueueItem(sound, channel!));
   }
 
   private onMessage(message: Message) {
     this.messageHandler.handle(message);
   }
 
   private onBotJoinsServer(guild: Guild) {
-    if (!guild || !guild.available) return;
+    if (!guild!.available) return;
```

Comment:
So either the parameter type is incorrect, or we do not need the bang here. Same issue on master actually, no need to check for guild if it is always defined, as the parameter type suggests.